atom/common/crash_reporter/crash_reporter_mac.mm

```
@@ -26,54 +50,54 @@
-
-  [parameters setValue:@ATOM_PRODUCT_NAME
-                forKey:@BREAKPAD_PRODUCT];
-  [parameters setValue:base::SysUTF8ToNSString(product_name)
-                forKey:@BREAKPAD_PRODUCT_DISPLAY];
-  [parameters setValue:base::SysUTF8ToNSString(version)
-                forKey:@BREAKPAD_VERSION];
-  [parameters setValue:base::SysUTF8ToNSString(company_name)
-                forKey:@BREAKPAD_VENDOR];
-  [parameters setValue:base::SysUTF8ToNSString(submit_url)
-                forKey:@BREAKPAD_URL];
-  [parameters setValue:(auto_submit ? @"YES" : @"NO")
-                forKey:@BREAKPAD_SKIP_CONFIRM];
-  [parameters setValue:(skip_system_crash_handler ? @"YES" : @"NO")
-                forKey:@BREAKPAD_SEND_AND_EXIT];
-
-  // Report all crashes (important for testing the crash reporter).
-  [parameters setValue:@"0" forKey:@BREAKPAD_REPORT_INTERVAL];
-
-  // Put dump files under "/tmp/ProductName Crashes".
+  static bool initialized = false;
+  if (initialized)
+    return;
+  initialized = true;
+
   std::string dump_dir = "/tmp/" + product_name + " Crashes";
-  [parameters setValue:base::SysUTF8ToNSString(dump_dir)
-                forKey:@BREAKPAD_DUMP_DIRECTORY];
-
-  // Temporarily run Breakpad in-process on 10.10 and later because APIs that
-  // it depends on got broken (http://crbug.com/386208).
-  // This can catch crashes in the browser process only.
-  if (base::mac::IsOSYosemiteOrLater()) {
-    [parameters setObject:[NSNumber numberWithBool:YES]
-                   forKey:@BREAKPAD_IN_PROCESS];
+  base::FilePath database_path(dump_dir);
+  if (is_browser_) {
+    @autoreleasepool {
+      base::FilePath framework_bundle_path = base::mac::FrameworkBundlePath();
+      base::FilePath handler_path =
+          framework_bundle_path.Append("Resources").Append("crashpad_handler");
+
+      crashpad::CrashpadClient crashpad_client;
+      // Send all crash reports.
+      std::vector<std::string> arguments = { "--upload-internal=0" };
```

My original motivation of adding `upload-internal` option is that this option will allow user to custom the upload rate control. Instead of hard-coding the rate, we can provide a `upload-internal` option in the `crashpad` API, how do you think it?